src/main/java/TaskManager.java

```diff
@@ -0,0 +1,318 @@
+            } catch (IncompleteCommandException e) {
+                MessagePrinter.incompleteCommand();
+            } catch (MissingKeyWordException e) {
+                MessagePrinter.missingKeyWord(e.getKeyword());
+            } catch (MissingDateException e) {
+                MessagePrinter.missingDate(e.getType());
+            }
+        }
+    }
+
+    /**
+     * Function to scan data from saved file upon initiating Friday
+     * Check if fle exists; if it doesn't, create new in main
+     * Works in loading from saved file
+     */
+    public static void loadData() throws FileNotFoundException {
+        // read from file using Scanner
+        File data = new File(FILEPATH);
+        Scanner s = new Scanner(data);
+        while (s.hasNext()) {
+            // read and parse data into Task array.
+            // Data stored in format type | isDone | taskname | date (if exists)
+            String[] splitString = s.nextLine().split("\\|");
+            boolean isDone = false;
+            if (splitString[1].trim().equals("X")) {
+                isDone = true;
+            }
+            try {
+                // if splitString only has 3 items; means its a todo
+                if (splitString.length == 3) {
+                    addToDo(splitString[0] + " " + splitString[2], isDone);
+                } else { // means it's a deadline or event
+                    if (splitString[0].trim().equals("deadline")) {
+                        addDeadline(splitString[0] + " " + splitString[2] + " /by " + splitString[3], isDone);
+                    } else {
+                        addEvent(splitString[0] + " " + splitString[2] + " /at " + splitString[3], isDone);
+                    }
+                }
+            } catch (EmptyTaskNameException e) {
+                MessagePrinter.emptyTaskName();
+            } catch (IncompleteCommandException e) {
+                MessagePrinter.incompleteCommand();
+            } catch (MissingKeyWordException e) {
+                MessagePrinter.missingKeyWord(e.getKeyword());
+            } catch (MissingDateException e) {
+                MessagePrinter.missingDate(e.getType());
+            }
+        }
+    }
+
+    /**
+     * @param textToAppend
+     * @throws IOException
+     * function to append new line when user adds in a new line
+     * Called in updateList to get live update as user enters
+     */
+    public static void appendToFile(String textToAppend) throws IOException {
+        FileWriter fw = new FileWriter(FILEPATH, true);
+        fw.write(textToAppend);
+        fw.close();
+    }
+
+    /**
+     * @param currTask: current task being added
+     * Function that appends task details in correct format in file
+     */
+    private static void updateList(Task currTask) {
+        String taskName = currTask.getTaskName();
+        String isDoneSymbol = currTask.isDone() ? "X" : "O";
+        // check prefix
+        if (currTask.getPrefix().equals("[T]")) {
```

Comment:
Perhaps the try catch need only surround the `appendToFile` function? The String within could be built first, following [Avoid Complicated Expressions](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions).